### PR TITLE
feat: My Collection artwork page submission status redesign

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Feature**

This PR solves [ONYX-1115]

- Review App: https://my-collection-artwork-page.artsy.net

### Description

This PR implements the new submission status section on the My Collection artwork page. 

It also includes a lot of refactoring, apologies for the huge PR 🙇 

> [!TIP]
> 💁 Please start your review [here](https://github.com/artsy/force/pull/14229/files/803cb57c5bc66a7e58e96cb0528c2fd6717e0434#diff-e767b3fc4af05fa8c95748027929ee64c584c098e0984b735e16a39a83fc46b9).

### Screenshots

**Submitted:**

| Desktop | mWeb | mWeb (What is this?) | 
| --- | --- | --- |
|Screenshot 2024-07-29 at 13 31 10 |Screenshot 2024-07-29 at 13 31 26 |Screenshot 2024-07-29 at 13 31 51 |

**Approved:**

| Desktop | mWeb | mWeb (What is this?) | 
| --- | --- | --- |
|Screenshot 2024-07-29 at 13 34 20 |Screenshot 2024-07-29 at 13 34 27 |Screenshot 2024-07-29 at 13 34 32|

[ONYX-1115]: https://artsyproduct.atlassian.net/browse/ONYX-1115?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASubmissionStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+          label: "Complete Submission",
+          to: `/sell/submissions/${submission.internalID}/${INITIAL_STEP}`,
+          variant: "primaryBlack",
+        },
+        description:
+          "You’ve started a submission to sell with Artsy but have not yet completed it.",
+        stateLabel: null,
+        actionLabel: "Complete Submission",
+      }
+    case "SUBMITTED":
```
Comment (dariakoko) [inline src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASubmissionStatus.tsx:254]: What do you think about adding this to MP? I see we already have some logic in ArtworkConsignmentSubmissionType

Comment (olerichter00) [reply src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASubmissionStatus.tsx:254]: Yes, good catch! We should definitely update and use the copy from MP. There should already be a ticket for it: https://artsy.slack.com/archives/C05EQL4R5N0/p1721746558213349?thread_ts=1721746117.454679&cid=C05EQL4R5N0

Comment (dariakoko) [reply src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASubmissionStatus.tsx:254]: Perfect 👍

File: src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASubmissionStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+      <Text variant="md" fontWeight="400">
+        {submission.stateLabel}
+      </Text>
+
+      <Media lessThan="sm">
+        {!!submission.actionLabel && !!buttonURL && (
+          <RouterLink
+            onClick={() => {
+              // TODO: Tracking
```
Comment (olerichter00) [inline src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASubmissionStatus.tsx:77]: https://artsyproduct.atlassian.net/browse/ONYX-1144

File: src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASubmissionStatus.tsx [hunk 3]
```diff
@@ -0,0 +1,246 @@
+import ChevronRightIcon from "@artsy/icons/ChevronRightIcon"
```
Comment (olerichter00) [inline src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASubmissionStatus.tsx:1]: This is the file containing the new submission status section on the MyC artwork page.

File: src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASubmissionStatus.tsx (hunk 1 above)
Comment (olerichter00) [reply src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASubmissionStatus.tsx]: done 👍

File: src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASubmissionStatus.tsx [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+  }
+
+  if (["APPROVED"].includes(submission.state)) {
+    return `/sell/submissions/${submission.internalID}/${INITIAL_POST_APPROVAL_STEP}`
+  }
+
+  return null
+}
+
+const getStateHelpMessage = (submission, isListed): JSX.Element => {
```
Comment (dariakoko) [inline src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASubmissionStatus.tsx:222]: What do you think about moving `getStateHelpMessage` and `getButtonURL` to utils?

**Review by dariakoko — APPROVED** [review]
Great work Ole!

**Pull request merged**